File: Sorting_Algorithms/SelectionSort.java
 import java.util.*;
 public class SelectionSort{
	 static int[] sorting(int[] a){
		int lengthOfArray=a.length;
		for(int i=0;i<lengthOfArray-1;i++){
			//int minimum=a[i];
			for(int j=i+1;j<lengthOfArray;j++){
				if(a[j]<a[i]){
					int temp=a[j];
					a[j]=a[i];
					a[i]=temp;
				}
			}
		}
		return a;
	}
	public static void main(String[] args){
		int[] unsortedArray=new int[]{12,3,15,22,89,32,45,16,8,90,55,14,78,63,23};
		int[] sortedArray=sorting(unsortedArray);
		System.out.print(Arrays.toString(sortedArray));
	}
}
/**
 * suppose the size of an array is n 
 * search a minimum element in range of [0,n-1] and swap it with index 0 element
 *  now search in range of [1,n-1] and swap it with index 1....
 */

File: Sorting_Algorithms/Selection_Sort.cpp
#include<iostream>
#include<array>
using namespace std;
int main() {
    array<int,10> Array={12,3,14,61,41,78,51,90,73,21};
    int length_of_array=Array.size();

    // sorting 
    for (int i = 0; i < length_of_array-1; i++){
        int minimun=Array[i];
        for (int j = i+1; j < length_of_array; j++){
            if(Array[j]<minimun){
                int temp=Array[i];
                Array[i]=Array[j];
                Array[j]=temp;
            }
        }
    }
    for(auto i : Array)
    {
        cout<<i<<"  ";
    }
    return 0;
}
/*
    find minimun from unsorted array and swap with 0 Index
    then 1 index
    then 2 .... and so on
*/